SmartDeviceLink/SDLMenuManager.m

```diff
@@ -112,6 +115,41 @@ - (void)stop {
 
 #pragma mark - Setters
 
+- (void)setMenuConfiguration:(SDLMenuConfiguration *)menuConfiguration {
+    if ([[SDLGlobals sharedGlobals].rpcVersion isLessThanVersion:[SDLVersion versionWithMajor:6 minor:0 patch:0]]) {
+        SDLLogW(@"Menu configurations is only supported on head units with RPC spec version 6.0.0 or later. Currently connected head unit RPC spec version is %@", [SDLGlobals sharedGlobals].rpcVersion);
+        return;
+    } else if (self.displayCapabilities.menuLayoutsAvailable == nil) {
+        SDLLogW(@"Could not set the main menu configuration. Which menu layouts can be used is not available");
+        return;
+    } else if (![self.displayCapabilities.menuLayoutsAvailable containsObject:menuConfiguration.mainMenuLayout]
+              || ![self.displayCapabilities.menuLayoutsAvailable containsObject:menuConfiguration.defaultSubmenuLayout]) {
+        SDLLogE(@"One or more of the set menu layouts are not available on this system. The menu configuration will not be set. Available menu layouts: %@, set menu layouts: %@", self.displayCapabilities.menuLayoutsAvailable, menuConfiguration);
+        return;
+    } else if (self.currentHMILevel == nil
+        || [self.currentHMILevel isEqualToEnum:SDLHMILevelNone]
+        || [self.currentSystemContext isEqualToEnum:SDLSystemContextMenu]) {
+        SDLLogE(@"Could not set main menu configuration, HMI level: %@, required: 'Not-NONE', system context: %@, required: 'Not MENU'", self.currentHMILevel, self.currentSystemContext);
```

Comment:
You should be able to change the menu layout when the menu is open. The description for the `menuLayout` parameter in the evolution proposal says: `Sets the layout of the main menu screen. If this is sent while a menu is already on-screen, the head unit will change the display to the new layout type.`